### PR TITLE
ADM-928:[frontend]refactor: update code style and add features

### PR DESCRIPTION
**Pull request by weiraneve**
## Summary

...

## Before

_Description_

**Screenshots**
If applicable, add screenshots to help explain behavior of your code.

## After

_Description_

**Screenshots**
If applicable, add screenshots to help explain behavior of your code.

## Note

_Null_

### CODE REVIEW

**Review by PengxiWPix — APPROVED** [review]
LGTM

File: frontend/__tests__/components/Common/DateRangeViewer/DateRangeViewer.test.tsx [hunk 1]
```diff
@@ -62,17 +63,23 @@ describe('DateRangeViewer', () => {
   });
 
   it('should show priority high icon when click expand button and step number is 1', async () => {
-    const failedTimeRanges = ['1706716800000'];
+    // given
```
Comment (mrcuriosity-tw) [inline frontend/__tests__/components/Common/DateRangeViewer/DateRangeViewer.test.tsx:66]: remove annotations

File: frontend/__tests__/components/Common/DateRangeViewer/DateRangeViewer.test.tsx [hunk 2]
```diff
@@ -62,17 +63,23 @@ describe('DateRangeViewer', () => {
   });
 
   it('should show priority high icon when click expand button and step number is 1', async () => {
-    const failedTimeRanges = ['1706716800000'];
```
Comment (Leiqiuhong) [inline frontend/__tests__/components/Common/DateRangeViewer/DateRangeViewer.test.tsx:65]: 这个修改为时间格式为年月日的那种吧，时间戳不直观。

File: frontend/src/hooks/useGetMetricsStepsEffect.ts [hunk 3]
```diff
@@ -21,6 +21,8 @@ export interface useGetMetricsStepsEffectInterface {
 }
 
 const TIMEOUT = 'timeout';
+const ERROR_STATUS = 'rejected';
```
Comment (mrcuriosity-tw) [inline frontend/src/hooks/useGetMetricsStepsEffect.ts:24]: better to put these 2 to `resources.ts`, besides, `fulfilled` and `rejected` are the 2/3(the another one is pending) key words of Promise A+ status. You can name them directly by `FULFILLED` & `REJECTED`.

File: frontend/src/context/stepper/StepperSlice.tsx [hunk 4]
```diff
@@ -5,15 +5,15 @@ import type { RootState } from '@src/store';
 export interface StepState {
   stepNumber: number;
   timeStamp: number;
-  shouldMetricsLoad: boolean;
-  failedTimeRange: string[];
+  shouldMetricsLoaded: boolean;
```
Comment (Leiqiuhong) [inline frontend/src/context/stepper/StepperSlice.tsx:8]: 这儿为什么需要修改shouldMetricsLoad的名称呢🤔

Comment (weiraneve) [reply frontend/src/context/stepper/StepperSlice.tsx:8]: zhongren's comment for shouldMetricsLoaded and shouldMetricsLoad

**Review by mrcuriosity-tw — APPROVED** [review]
LGTM

File: frontend/src/context/stepper/StepperSlice.tsx (hunk 4 above)
Comment (guzhongren) [reply frontend/src/context/stepper/StepperSlice.tsx:8]: Load 是动词，Loaded 是形容词 就这么简单 😄

**Pull request merged**